### PR TITLE
ddl: remove unused `copReqSenderPool` and related structures

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #54436

Problem Summary:

After https://github.com/pingcap/tidb/pull/54149, these code are only used by tests. Now I also migrate the test to disttask operator implementation, `copReqSenderPool` and related structures are completely unused.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

File: pkg/testkit/disttaskhelper/helper.go
Comment (D3Hunter) [inline pkg/testkit/disttaskhelper/helper.go:1]: maybe put into `ddl/testutil`?

**Pull request merged**